content/en/docs/concepts/workloads/controllers/replicaset.md

```diff
@@ -310,6 +310,36 @@ assuming that the number of replicas is not also changed).
 A ReplicaSet can be easily scaled up or down by simply updating the `.spec.replicas` field. The ReplicaSet controller
 ensures that a desired number of Pods with a matching label selector are available and operational.
 
+### Pod Deletion Cost 
+{{< feature-state for_k8s_version="v1.21" state="alpha" >}}
+
+Using the `controller.kubernetes.io/pod-deletion-cost` annotation, users can set a preference regarding 
+which pods to remove first when downscaling a ReplicaSet.
+
+The annotation should be set on the pod, and the expected type is `int32`. It represents the cost of
```

Comment:
I'm not sure how we define types in the API. There's the JSON API which doesn't think of numbers as having a number of bits, and then there are alternative representations on the wire.

We should define this the way that the API reference would.

Reply:
Tthe API reference does refer to the underlying type. See livenessProbe.initialDelaySeconds in https://kubernetes.io/docs/reference/kubernetes-api/workloads-resources/container/#lifecycle

Reply:
The issue is that using a larger int will cause the update to get rejected. So I think it is worth clarifying the expected size.

Reply:
> The API reference does refer to the underlying type

To be honest, exposing Golang details too much feels almost like a bug. We could say for example “an integer between 0 and 2<sup>32</sup>-1”.

Anyway, most people will guess it's unsigned I think, and we can tidy this up after v1.21 is released if we want to.

Reply:
actually it is signed, and negative values are allowed.

Reply:
@sftim are you still expecting more changes?

Reply:
Let's document the range of values. If this is -2147483648 to 2147483647 then let's say that, as well as mentioning the link to 2<sup>32</sup>. That way, every reader can infer that negative costs are permitted.

Reply:
this is not part of an object api, it is an annotation, should we add the type to reference/labels-annotations-taints.md ?

Reply:
Updated the text with the range and added the type to the reference and linked to it from here (see the paragraph above)